TaskListRails/app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,61 @@
+class TasksController < ApplicationController
+  def index
+  @tasks = Task.all
+
+  end
+  def show
+  @task = Task.find(params[:id])
+  end
+  def new
+  @task = Task.new
+  end
+  def create
+    @task = Task.new(task_params)
+
+    if @task.save
+    date_reset
+      redirect_to tasks_path
+    else
+      render :new
+    end
+  end
+  def edit
+    @task = Task.find(params[:id])
+  end
+
+  def update
+    @task = Task.find(params[:id])
+    if @task.update(task_params)
+    date_reset
+      redirect_to tasks_path
+    else
+      render :edit
+    end
+
+  end
+  def complete
```

Comment:
Ideally, you could have one line of whitespace between the `end` from the previous method and the `def` from the following method. Makes this much easier to read.